marine.c

```diff
@@ -588,15 +620,8 @@ WS_DLL_PUBLIC int marine_add_filter(char *bpf, char *dfilter, char **fields, int
 
     if (bpf != NULL) { // TODO add a function to validate bpfs
         has_bpf = TRUE;
-        pcap_t *pc;
-        pc = pcap_open_dead(DLT_EN10MB, MIN_PACKET_SIZE);
-        if (pc != NULL) {
-            if (pcap_compile(pc, &fcode, bpf, 0, 0) == -1) {
-                strcpy(err_msg, "Failed compiling the BPF");
-                pcap_close(pc);
-                return -1;
-            }
-            pcap_close(pc);
+        if (compile_bpf(bpf, &fcode, err_msg) == -1) {
```

Comment:
Unless the return value has actual meaning, I'd rather see `!= 0`

Reply:
See my comment on the booleans